### DISCUSSION TITLE
Context needed for translating a few strings to English

### DISCUSSION THREAD

**Discussion by ThrRip** [TTB-Network/python-openbmclapi#88] [全局]
Hi there @SilianZ,

Firstly, thank you for the great idea of localizing `python-openbmclapi`.

I joined the localization of this project today, and I've _mostly_ finished the translation of strings used for logging. However, there are a few where more context would help understand those strings and produce high-quality translations.

Currently there are 3 of them:
- `cluster.success.keepalive` ([Code](https://github.com/TTB-Network/python-openbmclapi/blob/7465f082d1ea19c178a951f9274ee9b2fef50a2a/i18n/en_us.json#L11), [Crowdin](https://crowdin.com/editor/python-openbmclapi-site/1660/zhcn-en?view=comfortable#69444))
- `cluster.error.download.failed` ([Code](https://github.com/TTB-Network/python-openbmclapi/blob/7465f082d1ea19c178a951f9274ee9b2fef50a2a/i18n/en_us.json#L30), [Crowdin](https://crowdin.com/editor/python-openbmclapi-site/1660/zhcn-en?view=comfortable#69482))
- `cluster.info.clear_cache.count` ([Code](https://github.com/TTB-Network/python-openbmclapi/blob/7465f082d1ea19c178a951f9274ee9b2fef50a2a/i18n/en_us.json#L44), [Crowdin](https://crowdin.com/editor/python-openbmclapi-site/1660/zhcn-en?view=comfortable#69510))

I'm kindly asking if you can provide some examples of these strings, with the variables (placeholders) in them replaced with actual data. Some simple explanation of when will these strings be included in the output would also be great.

Best regards,
ThrRip

**Comment by SilianZ** [comment]
Sure. And I want to know why you joined our Crowdin project.

**Reply by ThrRip** [reply]
I'm not sure if I understand your question. Are you willing to know how I found this project on Crowdin? Or something else?

**Reply by SilianZ** [reply]
Yes

**Reply by ThrRip** [reply]
That's a fairly simple question😄
![image](https://github.com/TTB-Network/python-openbmclapi/assets/54434330/e66ee8da-6cb4-496c-8e64-96418cc824a5)

**Comment by SilianZ** [comment]
> * `cluster.success.keepalive` ([Code](https://github.com/TTB-Network/python-openbmclapi/blob/7465f082d1ea19c178a951f9274ee9b2fef50a2a/i18n/en_us.json#L11), [Crowdin](https://crowdin.com/editor/python-openbmclapi-site/1660/zhcn-en?view=comfortable#69444))
> * `cluster.error.download.failed` ([Code](https://github.com/TTB-Network/python-openbmclapi/blob/7465f082d1ea19c178a951f9274ee9b2fef50a2a/i18n/en_us.json#L30), [Crowdin](https://crowdin.com/editor/python-openbmclapi-site/1660/zhcn-en?view=comfortable#69482))
> * `cluster.info.clear_cache.count` ([Code](https://github.com/TTB-Network/python-openbmclapi/blob/7465f082d1ea19c178a951f9274ee9b2fef50a2a/i18n/en_us.json#L44), [Crowdin](https://crowdin.com/editor/python-openbmclapi-site/1660/zhcn-en?view=comfortable#69510))

我已经在Crowdin的术语库上增添了几条对专有名词的解释和翻译，希望这能帮助你对文件进行翻译。

I have added several explanations and translations of proper nouns to Crowdin's glossary, hoping to help you translate documents.

**Reply by ThrRip** [reply]
Thank you very much.

I still have two simple questions though:
- In _回源了 `$hits` 个文件 (`$bytes`，`$count` 个储存)_, does _`$count` 个储存_ mean that files were served from `$count` storages?
- In _`$name` 过期缓存：`$count` (`$size`)。_, what does `$name` represent?

**Reply by SilianZ** [reply]
> * In _回源了 `$hits` 个文件 (`$bytes`，`$count` 个储存)_, does _`$count` 个储存_ mean that files were served from `$count` storages?

Yes, it is.

> * In _`$name` 过期缓存：`$count` (`$size`)。_, what does `$name` represent?

`$name` represents the name of the storage. Since users can configure multiple storage, so we use `$name` to represent each storage.

**Reply by SilianZ** [reply]
And there is a simple question. Why don't you use Chinese?

**Reply by ThrRip** [reply]
感谢回复！

---

> And there is a simple question. Why don't you use Chinese?

😉

---

最后一个小问题：

如果把 `Ping：$ping 毫秒。` 译为 `Ping: $pingms.`，会导致 `$ping` 无法被正常替换为实际数据吗？

**Reply by SilianZ** [reply]
> 感谢回复！
> 
> > And there is a simple question. Why don't you use Chinese?
> 
> 😉
> 
> 最后一个小问题：
> 
> 如果把 `Ping：$ping 毫秒。` 译为 `Ping: $pingms.`，会导致 `$ping` 无法被正常替换为实际数据吗？

会的，所以你应该用大括号将变量包裹起来：`${ping}ms`。

**Reply by ThrRip** [reply]
我刚刚根据你的修改，做了其他几处修改，比如 `node` -> `cluster`。

现在软件本身的 i18n 就完全完成了。

**Reply by SilianZ** [reply]
> 我刚刚根据你的修改，做了其他几处修改，比如 `node` -> `cluster`。
> 
> 现在软件本身的 i18n 就完全完成了。

万分感谢！
